src/Gaffer/ValuePlug.cpp

```diff
@@ -284,10 +307,32 @@ class ValuePlug::HashProcess : public Process
 				}
 
 				// And then look up the result in our cache.
-
+				
 				try
 				{
-					return threadData.cache.get( processKey );
+					if( g_hashCacheMode == HashCacheMode::Standard )
+					{
+						const HashProcessKey processKey( p, plug, Context::current(), p->m_dirtyCount, computeNode, cachePolicy );
+						return threadData.cache.get( processKey );
+					}
+					else if( g_hashCacheMode == HashCacheMode::Checked )
+					{
+						const HashProcessKey legacyProcessKey( p, plug, Context::current(), g_legacyGlobalDirtyCount + DIRTY_COUNT_RANGE_HALF, computeNode, cachePolicy );
+						IECore::MurmurHash check = threadData.cache.get( legacyProcessKey );
+						IECore::MurmurHash result = threadData.cache.get( processKey );
+
+						if( result != check )
+						{
+							throw IECore::Exception( "Failed to clear hash cache.  Fix affects() code." );
```

Comment:
I don't see how we can expect this Exception to get wrapped by a ProcessException for `plug`. Wrapping happens in `Process::handleException()`, but we're not inside a process for `plug` here - that process was created and destroyed inside `cache.get()`. That said, I haven't looked at your other example yet - going to take a look at that now.

Reply:
Wait, scratch that! I've expanded the surrounding code and seen the manual					`ProcessException::wrapCurrentException( processKey.plug, Context::current(), staticType );`